src/EnergyPlus/Plant/PlantManager.cc

```diff
@@ -4273,4 +4293,7 @@ void CheckOngoingPlantWarnings(EnergyPlusData &state)
 void EmptyPlantComponent::oneTimeInit([[maybe_unused]] EnergyPlusData &state)
 {
 }
+void EmptyPlantComponent::oneTimeInit_new([[maybe_unused]] EnergyPlusData &state)
```

Comment:
Interesting. Why do we have an `EmptyPlantComponent`?

Reply:
The initial purpose was so we could make [simulate](https://github.com/NREL/EnergyPlus/blob/feaf5115bb57b5f9f1890e4e9364312bb1c06752/src/EnergyPlus/PlantComponent.hh#L63) a pure virtual function. To do this, every plant component would need to have this function implemented.

Many of the [plant components did not have a comptr](https://github.com/NREL/EnergyPlus/blob/33b71ed598f37c10e9bfb6492e2a2103ba24e3dd/src/EnergyPlus/Plant/PlantManager.cc#L905). We created this EmptyPlantComponent that could be assigned to them ([example](https://github.com/NREL/EnergyPlus/blob/0d8c733ef731869e749c642c841642b89109730c/src/EnergyPlus/Plant/PlantManager.cc#L905)), so we could have [dummy simulate](https://github.com/NREL/EnergyPlus/blob/06e0eb6dfaac602c9faf65720a73c0d3cbdea256/src/EnergyPlus/Plant/PlantManager.hh#L112-L123) and oneTimeInit functions for them. Dummy functions would help us to not have any Diffs while we did this. I guess in the future, these components need to be scrutinized if the functions should be implemented separately. If yes, they need to be un-dummified and the functions implemented for every component that needs it.

Reply:
Works for me.

Also, how does one become un-dummified? Asking for a friend... 😉

Reply:
For humans, I wish I knew 😆